src/pybind/mgr/dashboard_v2/controllers/rbd.py

```diff
@@ -68,3 +95,35 @@ def get(self, pool_name):
         if status == ViewCache.VALUE_EXCEPTION:
             raise value
         return {'status': status, 'value': value}
+
+    def create(self, data):
+        if not self.rbd:
+            self.rbd = rbd.RBD()
+
+        # Get input values
+        name = data.get('name')
+        pool_name = data.get('pool_name')
+        size = data.get('size')
+        obj_size = data.get('obj_size')
+        features = data.get('features')
+        stripe_unit = data.get('stripe_unit')
+        stripe_count = data.get('stripe_count')
+        data_pool = data.get('data_pool')
+
+        # Set order
+        order = None
+        if obj_size and obj_size > 0:
+            order = int(round(math.log(float(obj_size), 2)))
+
+        # Set features
+        feature_bitmask = self._format_features(features)
+
+        ioctx = mgr.rados.open_ioctx(pool_name)
+
+        try:
+            self.rbd.create(ioctx, name, size, order=order, old_format=False,
+                            features=feature_bitmask, stripe_unit=stripe_unit,
+                            stripe_count=stripe_count, data_pool=data_pool)
+            return {'success': True, 'detail': 'Image {!s} created successfully'.format(name)}
+        except (InvalidArgument, ImageExists) as e:
```

Comment:
Why are you catching just these two exceptions?
I would catch `rbd.OSError`, which is the super class of rbd specific exceptions. Then I would also return to the frontend the `rbd.OSError.errno` integer code, so that we can cope with translation of error messages in the frontend.

Reply:
Yes, I agree.